### DISCUSSION TITLE
A way to expose logger.exception message first in Sentry

### DISCUSSION THREAD

**Discussion by warvariuc** [getsentry/sentry-python#2261] [Q&A]
I am logging an exception for a function like this:
```python
try:
    handler_obj.execute()
except Exception:  # pylint: disable=broad-except
    logger.exception("EventBridge handler %s error", handler_obj.__class__.__name__)

```
I provided a message to see the context of the exception in emails, Slack, etc., to know immediately where the particular exception happened.

Unfortunately, Sentry exposes the exception message first, hiding my message inside:

![image](https://github.com/getsentry/sentry-python/assets/1035452/9bab5e5a-e27e-4642-98e0-38b1974cdff7)

So I see `ConnectionError` in Slack, but this tells me nothing about the context.

Is there a way to expose the provided `EventBridge handler %s error` in Sentry ?

**Comment by TonisPiip** [comment]
I faced this same issue, and have this hack solution.
In slack integration you can choose which tags to include in the slack message.
But the logging message isn't included as a tag, so we manually add it to the tags in before_send, this seems to work.

Here's my example code. Oddly the set_tags doesn't seem to work in this scope, but maybe I was doing something wrong in testing.  

```
def before_send(event: Event, hint: Hint):

    # WARNING: seems sentry doesn't handle errors in this function well, so wrap everything in a try-except and test it before deploying
    # Set the logger message as a error tag so it can be put into the slack alert
    # this is a bit of a cludge, but set_tags doesn't seem to work here
    # DOC: https://docs.sentry.io/platforms/python/enriching-events/tags/

    message = event.get("logentry", {}).get("message", "")
    if message:
        # format the message with the logger params
        message = message % tuple(event.get("logentry", {}).get("params", []))
        tags = event["tags"] = event.get("tags", {})
        # Set and truncate the message to 200 characters
        tags.update({"message": message.replace("\n", "")[0:200]})
    
    return event

sentry_sdk.init(
        ...
        before_send=before_send,
        ...
    )
```

EDIT: you need to pass the formating parms as well if you want the message to be formated
